### PR TITLE
bugfix: Tooltip not enough funds when have enough balance

### LINKED ISSUE #3548: Tooltip not enough funds when have enough balance
Seems the tooltip about not having enough funds is poorly implemented, or proper checks are not there?

Screenshot 2022-07-24 at 07 25 38

https://user-images.githubusercontent.com/5887929/180633576-1bd15e9a-1179-42a8-8d58-4070ce29b05d.mov

This case might be ok but the tooltip is pointing elsewhere
Screenshot 2022-07-24 at 07 25 18

### PR DESCRIPTION
**Pull request by zhengow**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3548
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </bsx/gallery/2444376227-1>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

There should not have tooltip in owner actionList:
image

### CODE REVIEW

File: components/bsx/Gallery/Item/ActionList.vue [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <template>
   <div class="buttons">
     <b-tooltip
+      v-for="action in actions"
+      :key="action"
       class="w-100"
-      :active="!isMakeOffersAllowed"
+      :active="action === ShoppingActions.MAKE_OFFER && !isMakeOffersAllowed"
```
Comment (Jarsen136) [inline components/bsx/Gallery/Item/ActionList.vue:7]: This tooltip is not only used for 'Make Offer' action. 
Pls check it again.

image

Comment (zhengow) [reply components/bsx/Gallery/Item/ActionList.vue:7]: Actually if you look at old codes, it is only used for `Make Offer` action.
1. The tooltip component is only active when `isMakeOffersAllowed` prop is `false`, and 
2. The `isMakeOffersAllowed` prop has only one setter, which is related to `Make offer`:
```
this.isMakeOffersDisabled =
        !this.isMakeOffersAllowed || this.minimumOfferAmount > this.balance
```
which is in `AvailableActions.vue`'s `created` function
```
public async created(): Promise<void> {
    onApiConnect(() => {
      const { api } = Connector.getInstance()
      this.minimumOfferAmount = formatBsxBalanceToNumber(
        api?.consts?.marketplace?.minimumOfferAmount?.toString()
      )
      this.isMakeOffersDisabled =
        !this.isMakeOffersAllowed || this.minimumOfferAmount > this.balance

      if (!this.isMakeOffersAllowed) {
        this.tooltipOfferLabel = this.$t('tooltip.makeOfferDisabled')
      } else if (this.minimumOfferAmount > this.balance) {
        this.tooltipOfferLabel = this.$t('tooltip.makeOfferNotEnoughBalance')
      }
    })
  }
```
3. So what else action need to be used?

Comment (Jarsen136) [reply components/bsx/Gallery/Item/ActionList.vue:7]: Thank you for your explanation. You're right.

I have seen `action === ShoppingActions.MAKE_OFFER && !isMakeOffersAllowed` twice in this file. Let's wrap it with a get function. And it's good to go.

Comment (zhengow) [reply components/bsx/Gallery/Item/ActionList.vue]: done:)

**Review by cryptodamsky — COMMENTED** [review]
Got couple questions, currently at the time of screenshot, i have 0 funds.

So i can’t click on "Make Offer" which i would say is fine, but why the "You dont have enough funds" is placed over "Buy" button? It’s weird

image


And.. since i don’t have enough funds to make offer, why make the "Buy" button even clickable? Why does it even let me click it and trigger blockchain transaction when blockchain obviously cannot process the transaction because of low fees

image

**Comment by zhengow** [comment]
> And.. since i don’t have enough funds to make offer, why make the "Buy" button even clickable? Why does it even let me click it and trigger blockchain transaction when blockchain obviously cannot process the transaction because of low fees

Fix this problem, thanks for finding it.

> So i can’t click on "Make Offer" which i would say is fine, but why the "You dont have enough funds" is placed over "Buy" button? It’s weird
That's why this issue born, otherwise where it should be place? 👀

**Comment by cryptodamsky** [comment]
All i’m saying is that covering some parts, whether it is price, or another button with a tooltip looks bad from UX/UI standpoint. I guess it’s up for a discussion. 

Otherwise i can confirm that the option to click buy is gone now.

Btw, wouldn’t "Insufficient funds" sound better?

**Comment by zhengow** [comment]
> 

got it. I put tooltip in left, and disable `buy button` if it has insufficient funds. How about now?
image

**Review by cryptodamsky — COMMENTED** [review]
I think this looks way better now! 

The issue right now i have is that when i click on "buy now" and browsing gallery, when i go to certain nft for which i definitely have balance, the buy button is still disabled and showing insufficient funds?

image

This was found by gallery as " buy now "
[
For example this nft](https://deploy-preview-3551--koda-nuxt.netlify.app/bsx/gallery/3074475689-5?page=1)

**Comment by zhengow** [comment]
> 

Thank you, you found my bug!
The reason is that I only set `isBuyAllowed` in `created` function in `AvailableActions.vue`, which will not change when nft.price is queried.
Now I change it to reactivity, it will work now:)
image
image
image

**Review by cryptodamsky — APPROVED** [review]

**Review by Jarsen136 — APPROVED** [review]

**Comment by Jarsen136** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 56.3 USD/KSM ~ 1.776 $KSM
🧗 [EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws](https://kusama.subscan.io/account/EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)
🔗 [0x1c6d1fe3265f5961ec9916a2b3dfabdb5922a6877299b1c0fb9ca806feed7c0d](https://kusama.subscan.io/extrinsic/0x1c6d1fe3265f5961ec9916a2b3dfabdb5922a6877299b1c0fb9ca806feed7c0d)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**